clang/test/SemaCXX/warn-unused-private-field.cpp

```diff
@@ -20,6 +20,20 @@ class SpaceShipDefaultCompare {
   int operator<=>(const SpaceShipDefaultCompare &) const = default;
 };
 
+class EqDefaultCompareOutOfClass {
+  int used; // no warning
```

Comment:
Please explain a touch better with 'no warning'?  Something like, "Test to ensure that the defaulted operator results in this being seen as 'used' despite there not being code to do so."?

Reply:
Okay, sorry I wasn't sure what you were referring to, but I get know that from the change-set it is not obvious how / why the issue in #116961 is fixed by this change and why it is a special case not covered by the current testing. I expanded quite a bit on the comment, hopefully it did not become too verbose now.